docs/how-to/crafting/manage-dependencies.rst

```diff
@@ -3,4 +3,183 @@
+   which package is required.
+#. Add the package to your project file and repack the snap to see if you encounter
+   the same error.
+
+A typical missing build dependency will generate an error similar to the following:
+
+.. terminal::
+
+    configure: error: can't find the Boehm GC library.  Please install it.
+    Failed to run 'override-build': Exit code was 1.
+
+In most cases, the error will provide some indication of what needs to be installed. To
+resolve the error from the previous example output, the Boehm GC library (``libgc``)
+will need to be installed, and because this is the building stage, so too will its
+header package (``libgc-dev``).
+
+If the package is unknown, its correct name can normally be found with a search from
+within the build environment, a search engine, or on the `Ubuntu package archive
+<https://packages.ubuntu.com/>`_.
+
+
+Missing staging dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After a snap has been successfully built, if Snapcraft detects that the resultant
+executable will require further packages, it will attempt to guess these and output a
+list that can be copied and pasted into the snap's project file. The output will look
+similar to the following:
+
+.. terminal::
+
+    The 'example' part is missing libraries that are not included in the snap or base. They can be satisfied by adding the following entries to the existing stage-packages for this part:
+    - libxext6
+    - libxft2
+    - libxrender1
+    - libxss1
+
+To resolve the issue, copy the output list into the staging section of the referenced
+part.
+
+Another common problem is that even after snapcraft has successfully built a snap,
+running its executable will result in an error. The reasons for these errors are varied,
+but the most common is a missing library, as shown in the following example output:
+
+.. terminal::
+
+    /snap/mysnap/current/bin/mybin: error while loading shared libraries: libpaho-mqtt3a.so.1: cannot open shared object file: No such file or directory
+
+The following are the most common solutions for these kinds of errors:
+
+* If a required library might not have been installed by the snap, add the missing
+  package to the appropriate part's ``stage-packages`` key.
+* If the snap's ``LD_LIBRARY_PATH`` environment variable doesn't include the path to the
+  missing library, update ``LD_LIBRARY_PATH`` in the snap's project file. The following
+  example adds ``$SNAP/usr/lib``:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    apps:
+      example-app:
+        [...]
+        environment:
+          LD_LIBRARY_PATH: $LD_LIBRARY_PATH:$SNAP/usr/lib
+
+* If the missing library is shared to the snap by a content interface that isn't
+  properly connected, connect the snapd interface. This should be done manually for
+  testing and automatically for production.
+
+
+Remove stage package duplication
+--------------------------------
```

Comment:
@mr-cal I'd like your opinion on this:

```suggestion
Override a core dependency
--------------------------
```

Reply:
Oh, good catch.  I like your suggestion.

Reply:
[36d7fc3](https://github.com/canonical/snapcraft/pull/5452/commits/36d7fc39053fd35714292057809def4f57ce376b)